drivers/watchdog/wdt_nrf.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2018, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <nrfx_wdt.h>
+#include <nrf_clock.h>
+#include <clock_control.h>
+#include <watchdog.h>
+
+#define SYS_LOG_DOMAIN "WDT NRF"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_WDT_LEVEL
+#include <logging/sys_log.h>
+
+#define NRF_WDT_CHANNELS_NUMBER 8
+
+struct wdt_nrf_dev_data {
+	void (*cb)(struct device *dev, void *handle);
+};
+
+static struct device DEVICE_NAME_GET(wdt_nrf);
+
+/* There are eight reload registers in Nordic's WDT. */
+static nrfx_wdt_channel_id m_ch[NRF_WDT_CHANNELS_NUMBER];
+static wdt_callback_t      m_cb[NRF_WDT_CHANNELS_NUMBER];
+static u32_t               m_ch_id;
+
+
+static void wdt_event_handler(void)
+{
+	u32_t i;
+
+	for (i = 0; i < m_ch_id; ++i) {
+		if (nrf_wdt_request_status(m_ch[i])) {
+			if (m_cb[i]) {
+				m_cb[i](DEVICE_GET(wdt_nrf), &m_ch[i]);
+			}
+		}
+	}
+}
+
+static void wdt_nrf_isr(struct device *dev)
+{
+	ARG_UNUSED(dev);
+	/* We need to implement our own interrupt handler until nrfx one will
+	 * be fixed.
+	 * Clearing events also clears the reload register and has to be done
+	 * AFTER event handler.
+	 */
+	if (nrf_wdt_int_enable_check(NRF_WDT_INT_TIMEOUT_MASK)) {
+		wdt_event_handler();
+		nrf_wdt_event_clear(NRF_WDT_EVENT_TIMEOUT);
+	}
+}
+
+static int wdt_nrf_setup(struct device *dev, u8_t options)
+{
+	nrfx_wdt_config_t config;
+	nrfx_err_t err_code;
+
+	/* Set default value according to default value in MCU. */
+	config.behaviour = NRF_WDT_BEHAVIOUR_RUN_SLEEP;
+
+	ARG_UNUSED(dev);
+
+	if (!nrf_clock_lf_is_running()) {
+		return -ENOTSUP;
+	}
+
+	if (options & WDT_OPT_PAUSE_IN_SLEEP) {
+		config.behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_SLEEP;
+	}
+
+	if (options & WDT_OPT_PAUSE_HALTED_BY_DBG) {
+		config.behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_HALT;
+	}
+
+	config.reload_value = CONFIG_WDT_NRF_RELOAD_VALUE;
+
+	err_code = nrfx_wdt_init(&config, wdt_event_handler);
+	if (err_code == NRFX_ERROR_INVALID_STATE) {
+		return -EBUSY;
+	}
+
+	IRQ_CONNECT(CONFIG_WDT_NRF_IRQ,
+		CONFIG_WDT_NRF_IRQ_PRI, wdt_nrf_isr,
+		DEVICE_GET(wdt_nrf), 0);
+	irq_enable(CONFIG_WDT_NRF_IRQ);
+
+	nrfx_wdt_enable();
+
+	return 0;
+}
+
+static int wdt_nrf_disable(struct device *dev)
+{
+	/* Started watchdog cannot be stopped on nRF devices. */
+	ARG_UNUSED(dev);
+	return 0;
+}
+
+static int wdt_nrf_install_timeout(struct device *dev,
+				   struct wdt_timeout_cfg *cfg)
+{
+	ARG_UNUSED(dev);
+
+	nrfx_err_t err_code = nrfx_wdt_channel_alloc(&m_ch[m_ch_id]);
```

Comment:
Variable declaration always at the beginning of a code block, so here above ARG_UNUSED()